### PR TITLE
Cockpit 225 and c-podman 21

### PR DESCRIPTION
**Pull request by marusak**
Preview https://marusak.dev/cockpit-project.github.io/blog/cockpit-225.html

### CODE REVIEW

File: _posts/2020-08-05-cockpit-225.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+Libvirt supports two types of snapshots, [internal and external](https://libvirt.org/formatsnapshot.html),
+but external are not yet in feature parity with the internal. For this reason, cockpit developers decided to go
+with the internal snapshots in the Cockpit UI. Because of using internal snapshots this feature only works with
+qcow2 disk images. VM snapshots are supported for running VMs, in which case the snapshot will combine disk and
+VM memory state, as well as offline VMs, in which case only the disk contents will be saved.
+
+![VM snapshots](/images/machines-snapshots.png){:.screenshot}
+
+### Dev: Launching and reattaching to a long-running process
```
Comment (garrett) [inline _posts/2020-08-05-cockpit-225.md]:
```suggestion
### Developer API: Launch and reattach to a long-running process
```

File: _posts/2020-08-05-cockpit-225.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+which you can copy into your own project.
+
+### Podman: Add support for pods
+
+Alongside images and container now cockpit-podman plugin can also list pods with containers that belong
+into it. Also basic actions like starting or pausing pods are supported.
+
+TODO - screenshot if there is going to be delete button
+
+### Podman: Support checkpoint and restore operations for containers
```
Comment (garrett) [inline _posts/2020-08-05-cockpit-225.md:42]:
```suggestion
### Podman: Support checkpoint and restore
```

File: _posts/2020-08-05-cockpit-225.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+Checkpointing a container stops the container and writes the state of all container processes to disk.
+Such a checkpoint can be restored later to resume the container in the exact state as it was at checkpoint time.
+This capability requires [CRIU 3.11](https://www.criu.org/Main_Page) or later installed. Currently
+only the `runc` container runtime is supported; this is only available with `cgroupv1` systems.
+
+![Cockpit podman checkpoint](/images/podman-checkpoint.png){:.screenshot}
+
+_Thanks to Tomáš Glozar for this contribution!_
+
+### Podman: Enable selecting registries when downloading images
```
Comment (garrett) [inline _posts/2020-08-05-cockpit-225.md:53]:
```suggestion
### Podman: Registry selection in "download image" dialog
```

File: _posts/2020-08-05-cockpit-225.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+Users can pick specific registry to be used when looking for and downloading images.
+
+![Cockpit podman registries](/images/podman-registries.png){:.screenshot}
+
+### Podman: Enable untagging images
```
Comment (garrett) [inline _posts/2020-08-05-cockpit-225.md:59]:
```suggestion
### Podman: Selected tag removal during deletion
```

File: _posts/2020-08-05-cockpit-225.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+Cockpit pages should never directly run long-running precious processes such as Software installers or Ansible playbooks,
+as a Cockpit session is not reliably connected for a long time. If you work on such Cockpit page, please take a look
+at the new [long-running process example](https://github.com/cockpit-project/cockpit/tree/master/examples/long-running-process),
+which shows how to do that in a robust way: The process runs in a transient systemd unit, and the page reattaches
+to an already running process, so that the session can be terminated in between.
+
+The logic is implemented in a [JavaScript module](https://github.com/cockpit-project/cockpit/blob/master/pkg/lib/long-running-process.js)
+which you can copy into your own project.
+
+### Podman: Add support for pods
```
Comment (garrett) [inline _posts/2020-08-05-cockpit-225.md:35]:
```suggestion
### Podman: Support for pod groups
```

File: _posts/2020-08-05-cockpit-225.md (hunk 2 above)
Comment (garrett) [reply _posts/2020-08-05-cockpit-225.md:42]: (I figured "operations for containers" is a bit redundant with podman)

File: _posts/2020-08-05-cockpit-225.md (hunk 1 above)
Comment (garrett) [reply _posts/2020-08-05-cockpit-225.md]: This really isn't outdated. At least the stuff after `:`

**Comment by martinpitt** [comment]
bodhi URLs are all available now, both releases succeeded. :tada:

File: _posts/2020-08-05-cockpit-225.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly. Here are the release notes from Cockpit version 223 and Cockpit Podman version 21.
+
+### Machines: Add support for VM snapshots
+
+Libvirt supports two types of snapshots, [internal and external](https://libvirt.org/formatsnapshot.html),
+but external are not yet in feature parity with the internal. For this reason, cockpit developers decided to go
+with the internal snapshots in the Cockpit UI. Because of using internal snapshots this feature only works with
+qcow2 disk images. VM snapshots are supported for running VMs, in which case the snapshot will combine disk and
+VM memory state, as well as offline VMs, in which case only the disk contents will be saved.
```
Comment (garrett) [inline _posts/2020-08-05-cockpit-225.md:20]:
```suggestion
Cockpit now supports [libvirt's qcow2-based "internal" snapshotting](https://libvirt.org/formatsnapshot.html).

When a virtual machine is not running and Cockpit makes a snapshot, only the disk contents will be saved. When a virtual machine is running, Cockpit will snapshot both the disk and memory state.
```

Comment (garrett) [reply _posts/2020-08-05-cockpit-225.md:20]: Removed the comparison of internal versus external, which is too detailed for our audience and didn't make sense. (Yes, they're different _somehow_. But now? Yes, we chose one. Why? We didn't say, and I don't think it's relevant.)

I took the rest of the text and rewrote it.

File: _posts/2020-08-05-cockpit-225.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+### Developer API: Launch and reattach to a long-running process
+
+Cockpit pages should never directly run long-running precious processes such as Software installers or Ansible playbooks,
+as a Cockpit session is not reliably connected for a long time. If you work on such Cockpit page, please take a look
+at the new [long-running process example](https://github.com/cockpit-project/cockpit/tree/master/examples/long-running-process),
+which shows how to do that in a robust way: The process runs in a transient systemd unit, and the page reattaches
+to an already running process, so that the session can be terminated in between.
+
+The logic is implemented in a [JavaScript module](https://github.com/cockpit-project/cockpit/blob/master/pkg/lib/long-running-process.js)
+which you can copy into your own project.
```
Comment (garrett) [inline _posts/2020-08-05-cockpit-225.md:33]:
```suggestion
Pages in Cockpit should never tie long-running processes to the browser's session, as the page could reload at any time, the browser tab could be closed, or the computer running the browser could be shut down. These long-running sessions, such as software installers and Ansible playbooks, should persist across Cockpit sessions.

To anyone creating a Cockpit page, please review the new [long-running process example](https://github.com/cockpit-project/cockpit/tree/master/examples/long-running-process) to ensure your page is more robust. The example demonstrates how a to handle detatched sessions by running a process in a transient systemd unit and having the page attach (and re-attach, when the session changes) to the running process. Using this method allows for session interruptions.

All of this logic is implemented in a [JavaScript module](https://github.com/cockpit-project/cockpit/blob/master/pkg/lib/long-running-process.js) which we encourage you to copy into your own project.
```

File: _posts/2020-08-05-cockpit-225.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+### Podman: Support for pod groups
+
+Alongside images and container now cockpit-podman plugin can also list pods with containers that belong
+into it. Also basic actions like starting or pausing pods are supported.
```
Comment (garrett) [inline _posts/2020-08-05-cockpit-225.md:38]:
```suggestion
Initial support for pod groups in Cockpit has landed! The `cockpit-podman` plugin can now list pods (a group of containers) alongside the rest of the containers on a system. Basic actions like starting and pausing pods is also supported.
```

File: _posts/2020-08-05-cockpit-225.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+![Cockpit pod groups](/images/podman_pods.png){:.screenshot}
+
+### Podman: Support checkpoint and restore
+
+Checkpointing a container stops the container and writes the state of all container processes to disk.
+Such a checkpoint can be restored later to resume the container in the exact state as it was at checkpoint time.
+This capability requires [CRIU 3.11](https://www.criu.org/Main_Page) or later installed. Currently
+only the `runc` container runtime is supported; this is only available with `cgroupv1` systems.
```
Comment (garrett) [inline _posts/2020-08-05-cockpit-225.md:47]:
```suggestion
"Checkpointing" stops a container and writes the state of all container processes to disk. A container can later be resumed from a checkpoint in the exact state from when the checkpoint was made. 

This checkpointing capability requires [CRIU 3.11](https://www.criu.org/Main_Page) or later installed. Currently, only the `runc` container runtime is supported, which is only available with `cgroupv1` systems.
```

File: _posts/2020-08-05-cockpit-225.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+![Cockpit podman checkpoint](/images/podman-checkpoint.png){:.screenshot}
+
+_Thanks to Tomáš Glozar for this contribution!_
+
+### Podman: Registry selection in "download image" dialog
+
+Users can pick specific registry to be used when looking for and downloading images.
```
Comment (garrett) [inline _posts/2020-08-05-cockpit-225.md:55]:
```suggestion
When downloading images, a specific registry source can be selected.
```

File: _posts/2020-08-05-cockpit-225.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+![Cockpit podman registries](/images/podman-registries.png){:.screenshot}
+
+### Podman: Selected tag removal during deletion
+
+When deleting image that has multiple tags only specific tags can be picked to be removed.
```
Comment (garrett) [inline _posts/2020-08-05-cockpit-225.md:61]: Deleting images now shows a confirmation dialog where specific tags can be selected for removal.

**Comment by garrett** [comment]
The screenshots look good. I've optimized them (with pngquant and oxipng) to make them tiny in filesize.

**Comment by marusak** [comment]
thank you @garrett for the review and suggestions!
Incorporated them, also added urls for bodhi and squashed to one commit.

**Pull request merged**